frontend/src/components/GameCanvas.jsx

```diff
@@ -21,19 +21,27 @@ export default function GameCanvas({game}) {
   
     const [enemies, setEnemies] = useState([]);
     const [towers, setTowers] = useState([
+        castle,
+        portal,
         new ArrowTower(width-1, 0, heightMap.at(-1).at(0)),
         new RockTower(0, depth-1, heightMap.at(0).at(-1))
     ]);
```

Comment:
Not sure I like this. The common ancestor between the `Castle` and Towers is `Entity` so this can become a bit confusing.

Reply:
Yeah, it was a quick fix get the HpBar to update, the castle and portal are different. The castle may or may not attack, and we can see if the enemies will attack the tower.